network/src/peer_manager/mod.rs

```diff
@@ -423,12 +435,12 @@ where
             ConnectionRequest::DisconnectPeer(peer_id, resp_tx) => {
                 // Send a CloseConnection request to NetworkProvider and drop the send end of the
                 // NetworkRequest channel.
-                if let Some((_, _, sender)) = self.active_peers.remove(&peer_id) {
+                if let Some((conn_metadata, sender)) = self.active_peers.remove(&peer_id) {
                     // This should trigger a disconnect.
                     drop(sender);
                     // Add to outstanding disconnect requests.
                     self.outstanding_disconnect_requests
-                        .insert(peer_id, resp_tx);
+                        .insert(conn_metadata.connection_id, resp_tx);
```

Comment:
I see. At some point, we should explore this code and see if there are ways to make it a little more sane.